### DISCUSSION TITLE
`{if error}`, the try/catch(e) for templates

### DISCUSSION THREAD

**Discussion by robsonsobral** [ExpressionEngine/ExpressionEngine#2861] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Sometimes, specially on the **Search module**, we get messages using default system templates, which never matches the website design.

**Describe the solution you'd like**
Couldn't we have a conditional to be triggered on these cases?

**Describe alternatives you've considered**
＼（〇_ｏ）／

**Teachability, Documentation, Adoption, Migration Strategy**
We could check for `{if error}{/if}` on the tag data and return a new function `ee()->TMPL->error();` instead of `ee()->output->show_user_error()`.

```html
{exp:search:search_results}
    {if error}
        <h2>Error</h2>
        <p>{error_message}</p>
    {/if}

    {if no_results} Nothing found {/if}

    <h3>Results:</h3>

    <p>
        <a href="{auto_path}">{title}</a>
    </p>

    {paginate}
        <p>Page {current_page} of {total_pages} pages {pagination_links}</p>
    {/paginate}

{/exp:search:search_results}
```

- [x] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by robsonsobral** [comment]
I'm so dumb! Channel Form already has it! I just need to implement `error_handling="template"` parameter on Search Module!

**Comment by intoeetive** [comment]
I think this (`error_handling="inline"`) is something we need address on all forms in EE 

@TomJaeger I think we need make it supported on the level of functions `form_declaration`, `show_user_error` and the template parser. So that any add-on could use `errors` tag pair and inline error reporting, without necessity to custom code each time
